apps/core/scenes/search/Datepickers.js

```diff
@@ -2,141 +2,153 @@
-import { TripInput } from '@kiwicom/margarita-components';
-import {
-  Icon,
-  StyleSheet,
-  RangeDatePicker,
-} from '@kiwicom/universal-components';
-import { format } from 'date-fns';
+import { StyleSheet } from '@kiwicom/universal-components';
+import { format, isSameDay } from 'date-fns';
 import { defaultTokens } from '@kiwicom/orbit-design-tokens';
 import {
   withLayoutContext,
   LAYOUT,
   type LayoutContextState,
 } from '@kiwicom/margarita-utils';
-import { BASIC_ISO_DATE_FORMAT } from '@kiwicom/margarita-config';
+import { SHORT_DAY_MONTH_FORMAT } from '@kiwicom/margarita-config';
 
 import {
   withSearchContext,
   type TripTypes,
   type SearchContextState,
 } from './SearchContext';
 import PickersWrapper from './PickersWrapper';
+import DatePicker from './DatePicker';
 
 type Props = {|
   +tripType: string,
   +dateFrom: Date,
+  +dateTo: Date,
   +returnDateFrom: Date,
-  +setDepartureDate: Date => void,
-  +setReturnDate: Date => void,
+  +returnDateTo: Date,
+  +setDepartureDate: (Date, Date) => void,
+  +setReturnDate: (Date, Date) => void,
   +setTripType: TripTypes => void,
   +layout: number,
 |};
 
 type State = {|
-  isDatePickerVisible: boolean,
-  selectDate: $Keys<typeof DATEPICKER_MODE> | null,
+  isDepartureDatePickerVisible: boolean,
+  isArrivalDatePickerVisible: boolean,
 |};
 
-const DATEPICKER_MODE = {
-  DEPARTURE: 'DEPARTURE',
-  RETURN: 'RETURN',
-};
-
 class Datepickers extends React.Component<Props, State> {
   state = {
-    isDatePickerVisible: false,
-    selectDate: null,
+    isDepartureDatePickerVisible: false,
+    isArrivalDatePickerVisible: false,
   };
 
   handleDepartureDatePress = () => {
     this.setState({
-      isDatePickerVisible: true,
-      selectDate: DATEPICKER_MODE.DEPARTURE,
+      isDepartureDatePickerVisible: true,
     });
   };
 
   handleReturnDatePress = () => {
     this.setState({
-      isDatePickerVisible: true,
-      selectDate: DATEPICKER_MODE.RETURN,
+      isArrivalDatePickerVisible: true,
     });
   };
 
   handleDatePickerDismiss = () => {
     this.setState({
-      isDatePickerVisible: false,
+      isDepartureDatePickerVisible: false,
+      isArrivalDatePickerVisible: false,
     });
   };
 
-  handleDateChange = (date: Date) => {
-    const { selectDate } = this.state;
+  handleDateChange = (dates: Array<Date>) => {
     const {
       tripType,
       setDepartureDate,
       setReturnDate,
       setTripType,
     } = this.props;
 
-    switch (selectDate) {
-      case DATEPICKER_MODE.DEPARTURE:
-        setDepartureDate(date);
-        break;
-      case DATEPICKER_MODE.RETURN:
-        setReturnDate(date);
-        if (tripType === 'oneWay') {
-          setTripType('return');
-        }
-        break;
-      default:
-        break;
+    if (this.state.isDepartureDatePickerVisible) {
+      setDepartureDate(...dates);
+    }
+    if (this.state.isArrivalDatePickerVisible) {
+      setReturnDate(...dates);
+      if (tripType === 'oneWay') {
```

Comment:
Can we use the one exported from `@kiwicom/margarita-utils`?

Reply:
Fixed